addons/account/models/partner.py

```diff
@@ -677,3 +682,14 @@ def get_partner_localisation_fields_required_to_invoice(self, country_id):
         :return: an array of ir.model.fields for which the user should provide values.
         """
         return []
+
+    def _compute_days_sales_outstanding(self):
+        for partner in self:
```

Comment:
you use read_group
Warning amount_total is a field with currency, it is better to use amount_total_signed (computed in company currency).
And add ('company_id','=',self.env.company.id) in domain

and use fields.Date.conetxt_today(self), indeed fields.Date.today()